src/amr.it

```diff
@@ -0,0 +1,102 @@
+
+@sum Args begin
```

Comment:
Is this being copied from https://github.com/AlgebraicJulia/ACSets.jl/blob/main/src/ADTs.jl#L14?

We should upgrade that code to use InterTypes and then import that usage here.